library/init/meta/specialize_tactic.lean

```diff
@@ -0,0 +1,30 @@
+prelude
+
+import init.meta.tactic
+import init.meta.interactive_base
+import init.meta.interactive
+
+namespace tactic
+namespace interactive
+
+open interactive interactive.types expr
+
+meta def specialize_core (loc : expr) (args : list expr) : tactic unit :=
+do let original_name := loc.local_pp_name,
+   new_name ← mk_fresh_name,
+   rename original_name new_name,
+   F ← get_local new_name,
+   note original_name none (expr.mk_app F args),
+   clear_lst [new_name],
+   pure ()
+
+meta def specialize (e : parse texpr) : tactic unit := do
```

Comment:
Sounds good, was just waiting to see if people liked the implementation first :)